src/DotNetWorker.Grpc/FunctionMetadata/FunctionMetadataJsonReader.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.IO;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Script.Description;
+
+namespace Microsoft.Azure.Functions.Worker
+{ 
+    internal class FunctionMetadataJsonReader
+    {
+        private readonly string _directory;
+        private const string FileName = "functions.metadata";
+
+        public FunctionMetadataJsonReader(string directory)
+        {
+            _directory = directory ?? throw new ArgumentNullException(nameof(directory));
+        }
+
+        public virtual async Task<ImmutableArray<FunctionMetadata>> ReadMetadataAsync()
```

Comment:
I think that you should use modify this class to return the FunctionLoadRequest itself and drop the need for all of these extra types. Right now you're pulling it into the FunctionMetadata type, then converting it -- but it's not too complex; we can just update this to create the FunctionLoadRequest directly. It'd reduce dependencies and simplify things (and make them faster)...

I'd also recommend using System.Text.Json here rather than Newtonsoft.Json -- again, to reduce dependencies.

Reply:
Agreed. I think that will cleanup some additional things below as well.